From MAILER-DAEMON Mon Feb  5 18:10:02 2001
Date: 05 Feb 2001 18:10:02 +0100
From: Mail System Internal Data <MAILER-DAEMON@rudy.mif.pg.gda.pl>
Subject: DON'T DELETE THIS MESSAGE -- FOLDER INTERNAL DATA
X-IMAP: 0981393002 0000000000
Status: RO

This text is part of the internal format of your mail folder, and is not
a real message.  It is created automatically by the mail system software.
If deleted, important folder data will be lost, and it will be re-created
with the data reset to initial values.

From kloczek@rudy.mif.pg.gda.pl Mon Feb  5 05:32:26 2001 +0100
Status: R
X-Status: 
X-Keywords:
Return-Path: <owner-bugtraq@SECURITYFOCUS.COM>
Received: from venus.pk.edu.pl (qmailr@venus.wis.pk.edu.pl [149.156.131.32])
	by rudy.mif.pg.gda.pl (8.9.3/8.9.3) with SMTP id FAA04045
	for <kloczek@rudy.mif.pg.gda.pl>; Mon, 5 Feb 2001 05:32:26 +0100
Received: (qmail 4332 invoked by uid 501); 5 Feb 2001 04:32:27 -0000
Delivered-To: marcus@VENUS.WIS.PK.EDU.PL
Received: (qmail 4312 invoked from network); 5 Feb 2001 04:32:27 -0000
Received: from lists.securityfocus.com (66.38.151.7)
  by venus.wis.pk.edu.pl with SMTP; 5 Feb 2001 04:32:27 -0000
Received: from lists.securityfocus.com (lists.securityfocus.com [66.38.151.7])
	by lists.securityfocus.com (Postfix) with ESMTP
	id F32BA24D27D; Sun,  4 Feb 2001 21:01:23 -0700 (MST)
Received: from LISTS.SECURITYFOCUS.COM by LISTS.SECURITYFOCUS.COM
          (LISTSERV-TCP/IP release 1.8d) with spool id 24800115 for
          BUGTRAQ@LISTS.SECURITYFOCUS.COM; Sun, 4 Feb 2001 20:59:55 -0700
Approved-By: beng@SECURITYFOCUS.COM
Delivered-To: bugtraq@lists.securityfocus.com
Received: from securityfocus.com (mail.securityfocus.com [66.38.151.9]) by
          lists.securityfocus.com (Postfix) with SMTP id 7F8C224C405 for
          <bugtraq@lists.securityfocus.com>; Sun,  4 Feb 2001 01:24:40 -0700
          (MST)
Received: (qmail 10101 invoked by alias); 4 Feb 2001 08:24:47 -0000
Delivered-To: BUGTRAQ@SECURITYFOCUS.COM
Received: (qmail 10097 invoked from network); 4 Feb 2001 08:24:46 -0000
Received: from messi.uku.fi (193.167.224.8) by mail.securityfocus.com with
          SMTP; 4 Feb 2001 08:24:46 -0000
Received: (from jhuuskon@localhost) by messi.uku.fi (8.9.3/8.9.3) id KAA126428
          for BUGTRAQ@SECURITYFOCUS.COM; Sun, 4 Feb 2001 10:27:31 +0200
References: <008301c08b91$2df28080$501fb00a@cerc.dgaccp.pt>
            <20010202093629.A23686@ania.profnet.pl>
            <20010203155430.A3312@kerouac.mg.de>
Mime-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline
User-Agent: Mutt/1.2.5i
Message-ID:  <20010204102731.A100390@messi.uku.fi>
Date:         Sun, 4 Feb 2001 10:27:31 +0200
Reply-To: Jarno Huuskonen <Jarno.Huuskonen@UKU.FI>
Sender: Bugtraq List <BUGTRAQ@SECURITYFOCUS.COM>
From: Jarno Huuskonen <Jarno.Huuskonen@UKU.FI>
Subject:      Re: m4 format string vulnerability
To: BUGTRAQ@SECURITYFOCUS.COM
In-Reply-To:  <20010203155430.A3312@kerouac.mg.de>; from bluehell@GMX.NET on
              Sat, Feb 03, 2001 at 03:54:30PM +0100

On Sat, Feb 03, Mike Gerber wrote:
> confirmed for red hat linux 7.0:
>
> [kerouac:mg:~]m4 -G %x
> m4: 80499d9: Datei oder Verzeichnis nicht gefunden
> [kerouac:mg:~]cat /etc/redhat-release
> Red Hat Linux release 7.0 (Guinness)
> [kerouac:mg:~]rpm -q m4
> m4-1.4.1-3

I don't see this as a big problem, but here's a patch:

--- m4-1.4/src/m4.c.orig	Sat Feb  3 23:06:37 2001
+++ m4-1.4/src/m4.c	Sat Feb  3 23:07:26 2001
@@ -369,7 +369,7 @@

       case 'o':
 	if (!debug_set_output (optarg))
-	  error (0, errno, optarg);
+	  error (0, errno, "%s", optarg);
 	break;

       case 's':
@@ -466,7 +466,7 @@
 	    fp = path_search (argv[optind]);
 	    if (fp == NULL)
 	      {
-		error (0, errno, argv[optind]);
+		error (0, errno, "%s", argv[optind]);
 		continue;
 	      }
 	    else

One thing I noticed: if your system doesn't have mkstemp/tmpfile
m4 comes with its own implementation of tmpfile/mkstemp.
This uses mktemp and open (w/out O_EXCL).

-Jarno

